public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -980,20 +984,24 @@ else if(Integer.parseInt(ard.getStratum3()) == '8') {
 
         for(AchillesResult ar: aa.getResults()){
             String analysisStratumName=ar.getAnalysisStratumName();
-            if(ar.getStratum2().equals("2") && binRanges2.contains(Float.parseFloat(ar.getStratum4()))){
-                binRanges2.remove(Float.parseFloat(ar.getStratum4()));
-            }else if(ar.getStratum2().equals("3") && binRanges3.contains(Float.parseFloat(ar.getStratum4()))){
-                binRanges3.remove(Float.parseFloat(ar.getStratum4()));
-            }else if(ar.getStratum2().equals("4") && binRanges4.contains(Float.parseFloat(ar.getStratum4()))){
-                binRanges4.remove(Float.parseFloat(ar.getStratum4()));
-            }else if(ar.getStratum2().equals("5") && binRanges5.contains(Float.parseFloat(ar.getStratum4()))){
-                binRanges5.remove(Float.parseFloat(ar.getStratum4()));
-            }else if(ar.getStratum2().equals("6") && binRanges6.contains(Float.parseFloat(ar.getStratum4()))){
-                binRanges6.remove(Float.parseFloat(ar.getStratum4()));
-            }else if(ar.getStratum2().equals("7") && binRanges7.contains(Float.parseFloat(ar.getStratum4()))){
-                binRanges7.remove(Float.parseFloat(ar.getStratum4()));
-            }else if(ar.getStratum2().equals("8") && binRanges8.contains(Float.parseFloat(ar.getStratum4()))){
-                binRanges8.remove(Float.parseFloat(ar.getStratum4()));
+            try {
+                if(ar.getStratum2().equals("2") && binRanges2.contains(Float.parseFloat(ar.getStratum4()))){
+                    binRanges2.remove(Float.parseFloat(ar.getStratum4()));
+                }else if(ar.getStratum2().equals("3") && binRanges3.contains(Float.parseFloat(ar.getStratum4()))){
+                    binRanges3.remove(Float.parseFloat(ar.getStratum4()));
+                }else if(ar.getStratum2().equals("4") && binRanges4.contains(Float.parseFloat(ar.getStratum4()))){
+                    binRanges4.remove(Float.parseFloat(ar.getStratum4()));
+                }else if(ar.getStratum2().equals("5") && binRanges5.contains(Float.parseFloat(ar.getStratum4()))){
+                    binRanges5.remove(Float.parseFloat(ar.getStratum4()));
+                }else if(ar.getStratum2().equals("6") && binRanges6.contains(Float.parseFloat(ar.getStratum4()))){
+                    binRanges6.remove(Float.parseFloat(ar.getStratum4()));
+                }else if(ar.getStratum2().equals("7") && binRanges7.contains(Float.parseFloat(ar.getStratum4()))){
+                    binRanges7.remove(Float.parseFloat(ar.getStratum4()));
+                }else if(ar.getStratum2().equals("8") && binRanges8.contains(Float.parseFloat(ar.getStratum4()))){
+                    binRanges8.remove(Float.parseFloat(ar.getStratum4()));
+                }
+            } catch (NumberFormatException nfe) {
```

Comment:
Is this really expected? Why? Please put a comment here or log a warning if it's actually not expected.